### PR TITLE
Add support for alternate config file locations

### PR DESCRIPTION
**Pull request by GarethNelson**
Simple little patch, adds ability to select a different config file from the command line.

### CODE REVIEW

**Comment by madmaxoft** [comment]
This will not work yet, there's another mention of `"settings.ini"` in `cPluginManager::ReloadPluginsNow()`

File: src/Root.cpp [hunk 1]
```diff
@@ -141,8 +141,14 @@ void cRoot::Start(std::unique_ptr<cSettingsRepositoryInterface> a_OverridesRepo)
 
 	LOG("Reading server config...");
 
+	std::string conf_file = "settings.ini";
+	if (a_OverridesRepo->HasValue("Server","ConfigFile"))
+	{
+	conf_file = a_OverridesRepo->GetValue("Server","ConfigFile");
```
Comment (NiLSPACE) [inline src/Root.cpp]: The indentation is wrong here.

Comment (GarethNelson) [inline src/Root.cpp]: Can you elaborate on the indentation?
Also, @madmaxoft i'm working on PluginManager

**Comment by GarethNelson** [comment]
@NiLSPACE Please check src/Root.cpp again - I believe the indentation is now correct style

File: src/Root.cpp (hunk 1 above)
Comment (NiLSPACE) [inline src/Root.cpp]: You should add an extra tab for `conf_file = a_OverridesRepo..`

With the brace you're opening a new block, so by adding a new tab you're making the piece of code much more readable. This is especially more noticable if there is more code inside a block,

**Comment by NiLSPACE** [comment]
Yes, that's it :) Could you change over the `std::string` to `AString`?

**Comment by GarethNelson** [comment]
Working on it, next commit

**Comment by GarethNelson** [comment]
@NiLSPACE think this will be good once the CI tests pass?

File: src/Root.h [hunk 2]
```diff
@@ -58,6 +58,8 @@ class cRoot
 
 	void Start(std::unique_ptr<cSettingsRepositoryInterface> a_OverridesRepo);
 
+	AString m_SettingsFile;
```
Comment (madmaxoft) [inline src/Root.h]: We usually put variables first, then functions. So move this up.

Also, it's not a "file", it's a "filename".

Comment (madmaxoft) [inline src/Root.h]: And add a doxy-comment (`/** ... */`) explaining what this is, exactly.

Comment (GarethNelson) [inline src/Root.h]: See commit 25fe4a5 below

**Comment by GarethNelson** [comment]
@madmaxoft please review changes (and note for some reason travis is not passing despite showing as passed over on the travis site)

File: CONTRIBUTORS [hunk 3]
```diff
@@ -51,6 +51,7 @@ WebFreak001
 worktycho
 xoft (Mattes Dolak/madmaxoft on GH)
 Yeeeeezus (Donated AlchemistVillage prefabs)
+Gareth Nelson
```
Comment (madmaxoft) [inline CONTRIBUTORS]: Add yourself to the alpha-sorted position in the list.

**Comment by madmaxoft** [comment]
One more change and it should be good.

**Comment by GarethNelson** [comment]
@madmaxoft Done

**Comment by madmaxoft** [comment]
Thanks

**Pull request merged**